src/clims/services/extensible_service_api.py

```diff
@@ -103,3 +104,38 @@ def _get_filter_arguments(self, **kwargs):
             else:
                 get_args['archetype__{}'.format(key)] = value
         return get_args
+
+    @staticmethod
+    def _get_values_qs(property, extensible_type):
+        extensible_type_name = extensible_type.type_full_name_cls()
+        property_query_set = ExtensibleProperty.objects.filter(
+            extensible_property_type__extensible_type__name=extensible_type_name,
+            extensible_property_type__name=property).select_related('extensible_property_type')
+        return property_query_set
+
+    @staticmethod
+    def get_values_of_property(property, extensible_type):
+        property_query_set = ExtensibleServiceAPIMixin._get_values_qs(property, extensible_type)
+        try:
+            extensible_type = property_query_set[0].extensible_property_type
+            extensible_type_value_field = extensible_type.get_value_field()
+            values = property_query_set.all().values(extensible_type_value_field)
+            return ([x[extensible_type_value_field] for x in values])
+        except IndexError:
```

Comment:
If you change to not using an index, you would be able to catch `DoesNotExist` instead.